app/Helpers/Helpers.php

```diff
@@ -73,6 +73,11 @@ function cacheGet($cache_key)
     return Cache::get($cache_key);
 }
 
+function createCacheLock($cache_key)
+{
+    return Cache::lock($cache_key . '_lock'); 
```

Comment:
since we now know how the atomic lock works, adding five bytes to the "key"  could overflow memcache key limit. need to reduce the size here: https://github.com/faithcomesbyhearing/dbp/blob/e50f08651428aa3e23c44a6f1cf6c34687b312b6/app/Helpers/Helpers.php#L164 , from 250 to 245